.github/workflows/ci.yml

```diff
@@ -6,6 +6,10 @@ on:
 
 name: "CI"
 
+env:
+  # - COMPOSER_ROOT_VERSION is needed to get round the recursive dependency when using CI.
+  COMPOSER_ROOT_VERSION: '4.99.99'
```

Comment:
@jrfnl @sebastianbergmann 

Could also be resolved somewhat automatically, see https://github.com/ergebnis/composer-root-version-action.